src/main/java/io/reactiverse/pgclient/impl/codec/decoder/InitiateSslHandler.java

```diff
@@ -81,7 +79,9 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
       DecoderException err = (DecoderException) cause;
       cause = err.getCause();
     }
-    upgradeFuture.fail(cause);
+    if (!upgradeFuture.isComplete()) {
```

Comment:
rather use `upgradeFuture.tryFail(cause)`

Reply:
Updated in https://github.com/reactiverse/reactive-pg-client/pull/224/commits/d874dcb80c57141e5cdb7321dee348257753f29a.